apps/scan/src/main/scala/org/lfdecentralizedtrust/splice/scan/admin/api/client/BftScanConnection.scala

```diff
@@ -1080,57 +1175,160 @@ object BftScanConnection {
+                logger.info(
+                  s"Successfully established initial connection to trusted scan: ${scan.svName}"
+                )
+                Future.successful(Right(conn))
+              }
+              case Failure(err) => Future.successful(Left(scan.publicUrl -> err))
+            }
           )
-      case BftScanClientConfig.Bft(seedUrls, scansRefreshInterval, amuletRulesCacheTimeToLive) =>
-        for {
-          bft <- seedUrls
-            .traverse(uri =>
-              builder(uri, amuletRulesCacheTimeToLive).transformWith {
-                case Success(conn) => Future.successful(Right(conn))
-                case Failure(err) => Future.successful(Left(uri -> err))
+          (failed, connections) = initialConnections.toList.partitionEither(identity)
+
+          successfulConnectionDetails = connections
+            .map(c => s"  - ${c.config.adminApi.url}")
+            .mkString("\n")
+          failedConnectionDetails = failed
+            .map { case (uri, err) => s"  - $uri (${err.getMessage})" }
+            .mkString("\n")
+          _ = logger.info(
+            s"initial connection attempts complete. Successful (${connections.size}):\n$successfulConnectionDetails\nFailed (${failed.size}):\n$failedConnectionDetails"
+          )
+
+          scanList = new BftCustom(
+            ts.trustedSvs,
+            ts.threshold,
+            connections,
+            failed.toMap,
+            uri => builder(uri, ts.amuletRulesCacheTimeToLive),
+            Bft.getScansInDsoRules,
+            ts.scansRefreshInterval,
+            retryProvider,
+            loggerFactory,
+          )
+
+          bftConnection = new BftScanConnection(
+            spliceLedgerClient,
+            ts.amuletRulesCacheTimeToLive,
+            scanList,
+            clock,
+            retryProvider,
+            loggerFactory,
+          )
+
+          _ <- retryProvider.waitUntil(
+            RetryFor.WaitingOnInitDependency,
+            "refresh_initial_scan_list",
+            "Scan list is refreshed.",
+            scanList
+              .refresh(bftConnection)
+              .recoverWith { case NonFatal(ex) =>
+                Future.failed(
+                  Status.UNAVAILABLE
+                    .withDescription("Failed to refresh scan list on init")
+                    .withCause(ex)
+                    .asException()
+                )
               }
-            )
-            .map { cs =>
-              val (failed, connections) = cs.toList.partitionEither(identity)
-              new Bft(
-                connections,
-                failed.toMap,
-                uri => builder(uri, amuletRulesCacheTimeToLive),
-                Bft.getScansInDsoRules,
-                scansRefreshInterval,
-                retryProvider,
-                loggerFactory,
-              )
+              .map(_ => ()),
+            loggerFactory.getTracedLogger(classOf[BftScanConnection]),
+          )
```

Comment:
same as the point above, a lot of this code seems to be shared with the BftAny init so I would try to generalize it